coderd/workspaces.go

```diff
@@ -672,10 +672,11 @@ func (api *API) putWorkspaceTTL(rw http.ResponseWriter, r *http.Request) {
 	var dbTTL sql.NullInt64
 
 	err := api.Database.InTx(func(s database.Store) error {
+		var validityErr error
```

Comment:
Let me know if there's a better way to handle this - still getting the hang of Go!

Reply:
I think you can simplify this change to something like this:
```
		dbTTL, txErr := validWorkspaceTTLMillis(req.TTLMillis, 0)
		if txErr != nil {
			return codersdk.ValidationError{
				Field: "ttl_ms", 
				Detail: txErr .Error(),
			}
		}
		txErr = s.UpdateWorkspaceTTL(ctx, database.UpdateWorkspaceTTLParams{
			ID:  workspace.ID,
			Ttl: dbTTL,
		})
		if txErr != nil {
			return xerrors.Errorf("update workspace time until shutdown: %w", txErr )
		}
```

Reply:
I believe you'll need to change the `UpdateWorkspaceTTL` error as well to avoid this same problem.